src/descriptor/tr.rs

```diff
@@ -136,12 +138,30 @@ impl<Pk: MiniscriptKey, Ext: Extension> TapTree<Pk, Ext> {
         Q: MiniscriptKey,
         Ext: Extension,
     {
+        struct SimTranslator<'a, T>(&'a mut T);
+
+        impl<'a, Pk, T, Q, Error> simplicity::Translator<Pk, Q, Error> for SimTranslator<'a, T>
```

Comment:
In 899269acefeb534f140cfd4d7f86b66ad567b55f:

I wonder if, in rust-simplicity, we could just implement `SimTranslator` for all `Translator`s. Or if that'd break stuff.

Reply:
`elements_miniscript::Translator` and `miniscript::Translator` are distinct for some reason. If this doesn't clash with future plans, we could use `miniscript::Translator` across `elements_miniscript`, and implement `simplicity::Translator` for every `miniscript::Translator`.

I tried it out and the code compiled on my end.